cmd/config/docs/api-conventions/structured_output.md

```diff
@@ -0,0 +1,160 @@
+## Structured Output for Kubernetes CLI Tools
+Below is a protocol for emitting machine parseable structured output format (e.g. -o structured) from Kubernetes project tools. 
+This protocol would provide an API for better command integration & composition of tools in the Kubernetes ecosystem.
```

Comment:
Would you expand upon the motivation?  Maybe give a couple examples of where this would be useful by exec'ing commands, and how it facilitates composition of low-level tools by high-level tools?

Reply:
IIUC, this provides a io format for execing that is closer to http req/resp